device/iot-device-samples/multiplexing-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/MultiplexClientManager.java

```diff
@@ -0,0 +1,60 @@
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.device.IotHubConnectionStatusChangeCallback;
+import com.microsoft.azure.sdk.iot.device.MultiplexingClient;
+import lombok.experimental.Delegate;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+
+/**
+ * This class is in charge of handling reconnection logic and registering callbacks for connection status changes.
+ * It will delegate all other calls other than `Open`, `Close` and registerConnectionStatusChangeCallbaack to the inner client (MultiplexingClient)
+ */
+@Slf4j
+public class MultiplexClientManager extends ClientManagerBase {
+
+    // Define method calls that will not be delegated to the inner client.
+    private interface DeviceClientNonDelegatedFunction {
+        void open();
+        void close();
+        void registerConnectionStatusChangeCallback(IotHubConnectionStatusChangeCallback callback, Object callbackContext);
+    }
+
+    // The methods defined in the interface DeviceClientNonDelegatedFunction will be called on MultiplexingClientManager, and not on MultiplexingClient.
+    @Delegate(excludes = DeviceClientNonDelegatedFunction.class)
+    private final MultiplexingClient client;
+    private final String multiplexClientId;
+
+    /**
+     * Creates an instance of the MultiplexClientManager
+     *
+     * @param multiplexingClient the multiplexing client
+     * @param multiplexClientId user defined Id for the multiplexing client.
+     */
+    MultiplexClientManager(MultiplexingClient multiplexingClient, String multiplexClientId) {
+        this.connectionStatus = ConnectionStatus.DISCONNECTED;
+        this.client = multiplexingClient;
+        this.multiplexClientId = multiplexClientId;
```

Comment:
Where is this used?

Reply:
Just for logging, I guess?

Reply:
correct.

Reply:
Perhaps that makes me wonder why the multiplexing client doesn't have a way to uniquely identify it.

Anytime we have to write code in our samples (especially each time), I think that is a sign we are missing something in our client library. Do you agree?

Reply:
well, the question is how would you actually uniquely identify the multiplexing client? we either have to assign a random identifier to it as a metadata upon initialization, or take it as an input form the user. I don't know how much value there is in providing extra metadata to identify the multiplexing client ...

Reply:
The multiplexing client doesn't need to be identified to the service, but it would be for local use, as you've done with logging. I'd make it an optional, read-only property on the multiplexing client. If not provided, we could initialize it with a GUID - maybe not.

Reply:
I think at this stage since this is a sample code and it's only used for logging purposes taking it as a parameter in the constructor is reasonable. but I do agree that if at some point users start managing multiple multiplexingClients in their code, there might be a good argument that these clients should be easily identifiable which requires the core SDK logic to change (which is an addition and should be fine)